### PR TITLE
Remove logrus from validation and loader, update test warnings

### PR DESCRIPTION
**Pull request by edwardrf**
Compose warnings would be of the same format as the rest of the cli:
```
$ defang compose up --dry-run
 * Using AWS provider
 ! service "y": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "y": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "y": managed Redis service should use a redis image
 ! service "z": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "z": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "x": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "x": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "z": stateful service will lose data on restart; use a managed service instead
```

And a better diff which is a copy of gotools internal diff package, produces diff like this:
```
    --- FAIL: TestValidation/../../../tests/testproj/compose.yaml (0.01s)                                                                                                                                             
        validation_test.go:40: mismatch:                                                                                                                                                                              
            --- expected                                                                                                                                                                                              
            +++ actual                                                                                     
            @@ -1 +1 @@                                                                                  
            -level=warning msg="unsupported compose extension: \"x-unsupported\""                                                                                                                                     
            \ No newline at end of file                                                                                                                                                                               
            + ! unsupported compose extension: "x-unsupported"                                                                                                                                                        
            \ No newline at end of file                                                                                                                                                                               
                                                                                                                                                                                                                      
    --- FAIL: TestValidation/../../../tests/toomany/compose.yaml (0.01s)                  
        validation_test.go:40: mismatch:                                                                                                                                                                                                                                                                                                                                                                                                    
            --- expected                                                                                                                                                                                              
            +++ actual                                                                              
            @@ -1,2 +1,2 @@                                                                        
            -level=warning msg="service \"service1\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"
            -level=warning msg="service \"service1\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"                                                         
            \ No newline at end of file                                                                                                                                                                               
            + ! service "service1": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)                                                                                          
            + ! service "service1": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors                                                                            
            \ No newline at end of file                                                                                                                                                                               
                                                                                                                                                                                                                      
    --- FAIL: TestValidation/../../../tests/toomany/docker-compose.yml (0.01s)                                                                                                                                        
        validation_test.go:40: mismatch:                                                                                                                                                                              
            --- expected                                                                                                                                                                                              
            +++ actual                                                                                                                                                                                                
            @@ -1,2 +1,2 @@                                                          
            -level=warning msg="service \"service1\": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)"                                                                       
            -level=warning msg="service \"service1\": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors"                                                         
            \ No newline at end of file                                                                                                                                                                               
            + ! service "service1": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)                                                                                          
            + ! service "service1": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors                                                                            
            \ No newline at end of file    
```

### CODE REVIEW

File: src/pkg/cli/compose/validation.go [hunk 1]
```diff
@@ -280,8 +279,7 @@ func ValidateProject(project *compose.Project) error {
 }
 
 func warnf(format string, args ...interface{}) {
-	logrus.Warnf(format, args...)
-	term.SetHadWarnings(true)
+	term.Warnf(format, args...)
```
Comment (lionello) [inline src/pkg/cli/compose/validation.go:282]: should we still bother with this `warnf` function? Confusing.

File: src/pkg/logs/logrus.go [hunk 2]
```diff
@@ -17,3 +23,44 @@ func IsLogrusError(message string) bool {
...
+	case logrus.PanicLevel, logrus.FatalLevel:
+		f.Term.Fatal(getMessage())
+	case logrus.ErrorLevel:
+		f.Term.Error(getMessage())
+	case logrus.WarnLevel:
+		f.Term.Warn(getMessage())
+	case logrus.InfoLevel:
+		f.Term.Info(getMessage())
+	case logrus.DebugLevel, logrus.TraceLevel:
+		if f.Term.DoDebug() {
```
Comment (lionello) [inline src/pkg/logs/logrus.go:54]: no need for this `if`, `Debug` does that. Sure we save on `getMessage` but feels like premature optimization

File: src/pkg/logs/logrus.go [hunk 3]
```diff
@@ -17,3 +23,44 @@ func IsLogrusError(message string) bool {
...
+
+func (f TermLogFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	getMessage := func() string {
+		var buf strings.Builder
+		buf.WriteString(entry.Message)
+		for k, v := range entry.Data {
+			buf.WriteString(" ")
+			buf.WriteString(k)
+			buf.WriteString("=")
+			buf.WriteString(v.(string))
```
Comment (lionello) [inline src/pkg/logs/logrus.go:39]: is this safe?

**Review by lionello — CHANGES_REQUESTED** [review]

File: src/pkg/cli/compose/validation.go (hunk 1 above)
Comment (edwardrf) [reply src/pkg/cli/compose/validation.go]: Fixed

File: src/pkg/logs/logrus.go (hunk 2 above)
Comment (edwardrf) [reply src/pkg/logs/logrus.go]: Agreed

File: src/pkg/logs/logrus.go (hunk 3 above)
Comment (edwardrf) [reply src/pkg/logs/logrus.go]: Good catch, changed to use fmt.Fprintf with %v

**Pull request merged**